### PR TITLE
Check formulas for computing count

### PR DESCRIPTION
**Pull request by n-h-diaz**
Also moves out some formula functions into a separate package to avoid import cycles

I added the check to count rather than existence, since other apis also use count, so they'll also get the calculated observations added

### CODE REVIEW

File: internal/server/count/count.go [hunk 1]
```diff
@@ -132,3 +128,62 @@ func Count(
...
+		for _, entity := range entities {
+			if _, ok := result[dcid][entity]; ok {
+				continue
+			}
+			for _, f := range formulas {
+				variableFormula, err := formula.NewVariableFormula(f)
+				if err != nil {
+					return nil, err
+				}
+				calculatedCount, err := countInternal(
```
Comment (shifucun) [inline internal/server/count/count.go:164]: can you batch the countInternal call for all entities and variables?

Comment (n-h-diaz) [reply internal/server/count/count.go]: Done

File: internal/server/count/count.go [hunk 2]
```diff
@@ -132,3 +128,69 @@ func Count(
...
+	svOrSvgs []string,
+	entities []string,
+) (map[string]map[string]int32, error) {
+	result, err := countInternal(ctx, st, cachedata, svOrSvgs, entities)
+	if err != nil {
+		return nil, err
+	}
+	// Check for count for computed observations.
+	// Use counts of formula variables as a heuristic.
+	for _, dcid := range svOrSvgs {
```
Comment (shifucun) [inline internal/server/count/count.go:150]: nit: s/dcid/svOrSvg

File: internal/server/count/count.go [hunk 3]
```diff
@@ -132,3 +128,69 @@ func Count(
...
+		formulas, ok := cachedata.SVFormula()[dcid]
+		if !ok {
+			continue
+		}
+		missingEntities := []string{}
+		for _, entity := range entities {
+			if _, ok := result[dcid][entity]; !ok {
+				missingEntities = append(missingEntities, entity)
+			}
+		}
```
Comment (shifucun) [inline internal/server/count/count.go:160]: If the svOrSvg is a computed one, no entities would have the count data from countInternal? Should the following count happens when len(missingEntities) == len(entities)?

File: internal/server/count/count.go (hunk 2 above)
Comment (n-h-diaz) [reply internal/server/count/count.go]: Done

File: internal/server/count/count.go (hunk 3 above)
Comment (n-h-diaz) [reply internal/server/count/count.go]: Spoke offline - updated to check for all formulas in one call

**Review by shifucun — APPROVED** [review]
thanks for the update!

**Pull request merged**